tests/jerry/func-decl.js

```diff
@@ -38,3 +38,18 @@ check_syntax_error ("'use strict'; function arguments () {}");
 check_syntax_error ("'use strict'; var l = function arguments () {}");
 
 check_syntax_error ("function f__strict_mode_duplicate_parameters (p, p) { 'use strict'; }");
+
+function test_strict_mode_propagation_in_func_expr_and_getters_setters () {
```

Comment:
I was thinking about that what if future modifications/improvements a case occurs when this test passes and also some other cases passes when in should return an error for correct behavior.

Just like some parser tests when we create tests to accept correct input, but there are also tests to see if it does not parse some weird cases. BTW: maybe this does not need to be in the `fail` folder, but instead instead in some kind of try-catch.